src/os/JournalingObjectStore.cc

```diff
@@ -191,29 +193,35 @@ bool JournalingObjectStore::ApplyManager::commit_start()
   {
     Mutex::Locker l(apply_lock);
     dout(10) << "commit_start max_applied_seq " << max_applied_seq
-	     << ", open_ops " << open_ops
-	     << dendl;
+	     << ", open_ops " << open_ops << dendl;
+  
     blocked = true;
-    while (open_ops > 0) {
-      dout(10) << "commit_start waiting for " << open_ops << " open ops to drain" << dendl;
-      blocked_cond.Wait(apply_lock);
-    }
-    assert(open_ops == 0);
-    dout(10) << "commit_start blocked, all open_ops have completed" << dendl;
     {
       Mutex::Locker l(com_lock);
-      if (max_applied_seq == committed_seq) {
-	dout(10) << "commit_start nothing to do" << dendl;
+
+      set<uint64_t>::iterator it = applying_seq_set.begin();
+      if (it == applying_seq_set.end() ){
+       	_committing_seq = committing_seq = max_applied_seq; 
+		
+      dout(10) << "commit_start committing from end: " << committing_seq 
+		<< ",current applying_seq_set size:"<< applying_seq_set.size()
+		<< ", still blocked" << dendl;
+		assert( applying_seq_set.size() == 0 );
+      }else
+      {
+        _committing_seq = committing_seq = *it - 1;
```

Comment:
Because of the apply operation are called by OpWq, which  has multi threads.The apply operations are paralleled by multi threads. if use max_applied_seq directly, it doesn't make sure all the appply operation  which may has smaller seq  has completed.

In old way, it do use max_applied_seq directly, but it must make sure the tp thread pool are paused and wait all the inflight apply operation completed.
In my way, it  doesn't  pause the tp thread pool and doesn't wait all the inflight apply operaton completed, it just use the smalled seq that has completed to sync,  leaving inflight apply operations to be synced next time.